### PR TITLE
helm: default 2fa to `on` instead of `otp`

### LINKED ISSUE #19394: `teleport-cluster` Helm chart should support webauthn and passwordless by default
What would you like Teleport to do?

Clusters deployed using the `teleport-cluster` Helm chart should support webauthn and passwordless by default instead of supporting only OTP. We can do this by default in the Helm chart because we know the public address and can automatically set the `rp_id` (which is not the case when deploying Teleport using the binary only).

What problem does this solve?

This allows users to get more value from Teleport by default, as they can log in using hardware tokens.

If a workaround exists, please include it.

Learn about the 2fa configuration and the chart parameters, and set the value ~`authentication.second_factor: webauthn`~ `authentication.second_factor: on`.

### PR DESCRIPTION
**Pull request by hugoShaka**
Fixes https://github.com/gravitational/teleport/issues/19394

This change adds support for FIDO2/webauthn/hardware tokens by default. OTP 2fa will remain functional. This is a major change and should be part of a major release, even if this should be seamless for users. I would like for this change to be part of v12.

Warning: `webauthn.rp_id` should not change in the cluster life, else 2fa tokens will have to be re-registered.

Users accessing the cluster under a different name than `clusterName` will have to set rp_id
(`auth.teleportConfig.auth_service.authentication.webauthn.rp_id`) to be able to register second factors. As we strongly encourage users to have a resolvable `clusterName` and `publicAddr` support got added recently, this seems an acceptable edge case.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

**Comment by zmb3** [comment]
Tagged with `release-notes` so we can mention this in the v12 release notes.

File: docs/pages/reference/helm-reference/teleport-cluster.mdx [hunk 1]
```diff
@@ -116,13 +116,31 @@ Defaults to Teleport's binary default when empty: `best_effort`.
...
+than [`clusterName`](#clusterName) you must manually set the webauthn [Relying
+Party Identifier (RP
+ID)](https://www.w3.org/TR/webauthn-2/#relying-party-identifier). If you don't,
+RP ID will default to `clusterName` and users will fail to register second
+factors.
+
+You can do this by setting the value
+`auth.teleportConfig.auth_service.authentication.webauthn.rp_id`.
+
+RP ID must be valid domain, part of the full domain users are connecting to.
```
Comment (webvictim) [inline docs/pages/reference/helm-reference/teleport-cluster.mdx:138]:
```suggestion
RP ID must be both a valid domain, and part of the full domain users are connecting to.
```

File: docs/pages/reference/helm-reference/teleport-cluster.mdx [hunk 2]
```diff
@@ -116,13 +116,31 @@ Defaults to Teleport's binary default when empty: `best_effort`.
...
 are `off` (not recommended), `on`, `otp`, `optional` and `webauthn`.
 
-When set to `on`, `optional` or `webauthn`, the `authenticationSecondFactor.webauthn` section can also be used. The configured `rp_id` defaults to
-the FQDN which is used to access the Teleport cluster cluster-wide.
+When set to `on`, `optional` or `webauthn`, the `authenticationSecondFactor.webauthn` section can also be used.
+The configured `rp_id` defaults to `clusterName`.
+
+<Admonition type="warning">
+If you set `publicAddr` for users to access the cluster under a domain different
+than [`clusterName`](#clusterName) you must manually set the webauthn [Relying
```
Comment (webvictim) [inline docs/pages/reference/helm-reference/teleport-cluster.mdx:129]:
```suggestion
to [`clusterName`](#clusterName), you must manually set the webauthn [Relying
```

File: docs/pages/reference/helm-reference/teleport-cluster.mdx [hunk 3]
```diff
@@ -271,6 +289,24 @@ These settings will not apply if [`proxyListenerMode`](#proxylistenermode) is se
 
 When `publicAddr` is not set, the address used is [`clusterName`](#clusterName) on port 443.
 
+<Admonition type="warning">
+If you set `publicAddr` for users to access the cluster under a domain different
+than [`clusterName`](#clusterName) you must manually set the webauthn [Relying
```
Comment (webvictim) [inline docs/pages/reference/helm-reference/teleport-cluster.mdx:294]:
```suggestion
to [`clusterName`](#clusterName) you must manually set the webauthn [Relying
```

File: docs/pages/reference/helm-reference/teleport-cluster.mdx [hunk 4]
```diff
@@ -271,6 +289,24 @@ These settings will not apply if [`proxyListenerMode`](#proxylistenermode) is se
...
+Party Identifier (RP
+ID)](https://www.w3.org/TR/webauthn-2/#relying-party-identifier). If you don't,
+RP ID will default to `clusterName` and users will fail to register second
+factors.
+
+You can do this by setting the value
+`auth.teleportConfig.auth_service.authentication.webauthn.rp_id`.
+
+RP ID must be valid domain, part of the full domain users are connecting to.
```
Comment (webvictim) [inline docs/pages/reference/helm-reference/teleport-cluster.mdx:303]:
```suggestion
RP ID must be both a valid domain, and part of the full domain users are connecting to.
```

File: examples/chart/teleport-cluster/values.yaml [hunk 5]
```diff
@@ -63,7 +63,24 @@ authentication:
...
   # Possible values are 'off', 'on', 'otp', 'optional' and 'webauthn'.
-  secondFactor: "otp"
+  #
+  # WARNING:
+  #   If you set `publicAddr` for users to access the cluster under a domain different
+  #   than [`clusterName`](#clusterName) you must manually set the webauthn [Relying
+  #   Party Identifier (RP
+  #   ID)](https://www.w3.org/TR/webauthn-2/#relying-party-identifier). If you don't,
+  #   RP ID will default to `clusterName` and users will fail to register second
+  #   factors.
```
Comment (webvictim) [inline examples/chart/teleport-cluster/values.yaml:73]:
```suggestion
  #   to clusterName you must manually set the webauthn Relying
  #   Party Identifier (RP ID) - https://www.w3.org/TR/webauthn-2/#relying-party-identifier
  #   If you don't, RP ID will default to `clusterName` and users will fail
  #   to register second factors.
```

I don't think anchors/links will work in this file and the formatting looks weird

File: examples/chart/teleport-cluster/values.yaml [hunk 6]
```diff
@@ -109,6 +126,23 @@ separateMongoListener: false
...
 # Do not set any of these values unless you explicitly need to. Teleport always uses the cluster name by default.
 # Public cluster addresses, including port. Defaults to `clusterName` on 443 port.
+#
+# WARNING:
+#   If you set `publicAddr` for users to access the cluster under a domain different
+#   than [`clusterName`](#clusterName) you must manually set the webauthn [Relying
+#   Party Identifier (RP
+#   ID)](https://www.w3.org/TR/webauthn-2/#relying-party-identifier). If you don't,
+#   RP ID will default to `clusterName` and users will fail to register second
+#   factors.
```
Comment (webvictim) [inline examples/chart/teleport-cluster/values.yaml:136]:
```suggestion
  #   to clusterName you must manually set the webauthn Relying
  #   Party Identifier (RP ID) - https://www.w3.org/TR/webauthn-2/#relying-party-identifier
  #   If you don't, RP ID will default to `clusterName` and users will fail
  #   to register second factors.
```

Ditto above

File: examples/chart/teleport-cluster/values.yaml [hunk 7]
```diff
@@ -63,7 +63,24 @@ authentication:
...
+  #   than [`clusterName`](#clusterName) you must manually set the webauthn [Relying
+  #   Party Identifier (RP
+  #   ID)](https://www.w3.org/TR/webauthn-2/#relying-party-identifier). If you don't,
+  #   RP ID will default to `clusterName` and users will fail to register second
+  #   factors.
+  # 
+  #   You can do this by setting the value
+  #   `auth.teleportConfig.auth_service.authentication.webauthn.rp_id`.
+  #   
+  #   RP ID must be valid domain, part of the full domain users are connecting to.
```
Comment (webvictim) [inline examples/chart/teleport-cluster/values.yaml:78]:
```suggestion
  #   RP ID must be both a valid domain, and part of the full domain users are connecting to.
```

File: examples/chart/teleport-cluster/values.yaml [hunk 8]
```diff
@@ -109,6 +126,23 @@ separateMongoListener: false
...
+#   than [`clusterName`](#clusterName) you must manually set the webauthn [Relying
+#   Party Identifier (RP
+#   ID)](https://www.w3.org/TR/webauthn-2/#relying-party-identifier). If you don't,
+#   RP ID will default to `clusterName` and users will fail to register second
+#   factors.
+# 
+#   You can do this by setting the value
+#   `auth.teleportConfig.auth_service.authentication.webauthn.rp_id`.
+#   
+#   RP ID must be valid domain, part of the full domain users are connecting to.
```
Comment (webvictim) [inline examples/chart/teleport-cluster/values.yaml:141]:
```suggestion
#   RP ID must be both a valid domain, and part of the full domain users are connecting to.
```

**Review by webvictim — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Review by marcoandredinis — APPROVED** [review]

**Pull request merged**